ssl/record/rec_layer_d1.c

```diff
@@ -686,6 +686,23 @@ int dtls1_read_bytes(SSL *s, int type, int *recvd_type, unsigned char *buf,
         goto start;
     }
 
+    /*
+     * If we are a server and get a client hello when renegotiation isn't
+     * allowed send back a no renegotiation alert and carry on.
+     */
+    if (s->server
+            && SSL_is_init_finished(s)
+            && !s->s3->send_connection_binding
+            && s->rlayer.d->handshake_fragment_len >= 4
```

Comment:
side-note: what's the meaning of 4?  reneg message len?

Reply:
Good question. This is the size of a handshake header message. Probably the 4 here should be `SSL3_HM_HEADER_LENGTH`.  Mind if I change it before I push?

Reply:
please do.  and in the other PR's too, if applicable.  reconfirmed.